exc/types.go

```diff
@@ -0,0 +1,83 @@
+package exc
+
+import (
+	"errors"
+	"strconv"
+)
+
+// Type indicates the type of error, mirroring those in rpc.capnp.
+type Type int
```

Comment:
I don't like re-declaring this by hand vs. just importing the generated code. Esp. with stuff like the String() method, which `rpc.Exception_Type` already has...

Reply:
Oh, I didn't realize the generated code already had this.  Is there a reason it wasn't used in the original `internal/errors` package?

Reply:
Ah, I didn't even notice you'd just moved it when making this comment. @zombiezen would have to weigh in wrt if there was a specific reason not to, but the only possible reason I see is that the names are a bit more verbose. But if nothing else I thing we should make these aliases...

Let's put this off until after this pr is merged though; it's already a bit off-topic. But since we're exporting the package, we should be intentional about this before we stabilize the API.

Reply:
It seems to introduce an import cycle too, so maybe that's the reason.

Just to be clear:  are you saying I should revert the changes to the error package or simply hold off on aliasing `Type` with `Exception_Type`? I'd prefer to at least keep the present changes, unless you strongly object.  They make it quite a bit easier to handle errors internally, which is quite helpful.

Reply:
Just hold off on aliasing.